Tests/SQLiteNIOTests/SQLiteNIOTests.swift

```diff
@@ -26,6 +26,8 @@ final class SQLiteNIOTests: XCTestCase {
         let rows = try conn.query("SELECT ? as date", [date.sqliteData!]).wait()
         XCTAssertEqual(rows[0].column("date"), .float(date.timeIntervalSince1970))
         XCTAssertEqual(Date(sqliteData: rows[0].column("date")!)?.description, date.description)
+        XCTAssertEqual(Date(sqliteData: rows[0].column("date")!), date)
+        XCTAssertEqual(Date(sqliteData: rows[0].column("date")!)?.timeIntervalSinceReferenceDate, date.timeIntervalSinceReferenceDate)
```

Comment:
Instead of relying on repeated test runs to demonstrate the correctness of the behavior, can you:

1. Leave these additions to this test in place, these assertions are good to have anyhow.
2. Run the test against a version _without_ your fix until you come across a particular value which triggers the failure (or if you already know a specific value that does so, just grab that, of course 😅). If you by chance happen to find or already have more than one such value, even better!
3. Add a new test which uses the affected value(s) specifically, rather than `Date()`, so failure of the old code and success of your fixed version are visible and deterministic.
4. Finally, add another new test which verifies that timestamps are always clamped to µs precision - start with a higher precision, roundtrip through `SQLiteDataConvertible` (you don't need to do it via a query, really)  and assert that only microseconds are retained, including being rounded up or down as appropriate.

Reply:
Awesome suggestion! I have added these tests.